reaction.go

```diff
@@ -1,27 +1,78 @@
 package stream_chat
 
+import (
+	"errors"
+	"net/http"
+	"net/url"
+	"path"
+)
+
 type Reaction struct {
 	MessageID string `json:"message_id"`
-	UserID    string `json:"user_id,omitempty"`
+	UserID    string `json:"user_id"`
 	Type      string `json:"type"`
 
 	// any other fields the user wants to attach a reaction
-	ExtraData map[string]interface{}
+	ExtraData map[string]interface{} `json:"-,extra"`
+}
+
+type reactionResponse struct {
+	Message  Message  `json:"message"`
+	Reaction Reaction `json:"reaction"`
+}
+
+type reactionRequest struct {
+	Reaction *Reaction `json:"reaction"`
+}
+
+// SendReaction sends a reaction about a message
+//
+// message: pointer to the message struct
+// reaction: the reaction object, ie {type: 'love'}
+// userID: the ID of the user that created the reaction
+func (ch *Channel) SendReaction(msg *Message, reaction *Reaction, userID string) error {
```

Comment:
Not clear that the message is a result destination.

Reply:
what is not clear?

Reply:
Client modifies msg. That could be unexpected. Simple godoc comment will do fine.